tests/SetCookieTest.php

```diff
@@ -89,5 +90,96 @@ public function test_it_converts_to_header_value()
         $expiresInFiveYear = time() + 5 * 365 * 3600 * 24;
         $expected = sprintf('name=value; expires=%s; path=/path/; domain=domain.tld', gmdate(self::$HTTP_DATE_FORMAT, $expiresInFiveYear));
         $this->assertEquals($expected, $cookie->toHeaderValue());
+
+        $expireTS = time();
+        $now = new DateTimeImmutable("@{$expireTS}");
+        $cookie = SetCookie::thatExpiresAt('name', 'value', $now);
+        $expected = sprintf('name=value; expires=%s', gmdate(self::$HTTP_DATE_FORMAT, $expireTS));
+        $this->assertEquals($expected, $cookie->toHeaderValue());
+
+        // test positive seconds interval
+        $expireSecs = 123;
+        $expireTS = time() + $expireSecs;
+        $cookie = SetCookie::thatExpiresIn('name', 'value', $expireSecs);
+        $expected = sprintf('name=value; expires=%s', gmdate(self::$HTTP_DATE_FORMAT, $expireTS));
+        $this->assertEquals($expected, $cookie->toHeaderValue());
+
+        // test negative seconds interval
+        $expireSecs = -123;
+        $expireTS = time() + $expireSecs;
+        $cookie = SetCookie::thatExpiresIn('name', 'value', $expireSecs);
+        $expected = sprintf('name=value; expires=%s', gmdate(self::$HTTP_DATE_FORMAT, $expireTS));
+        $this->assertEquals($expected, $cookie->toHeaderValue());
+
+        // test positive string expression interval
+        $expireStr = '1 day';
+        $expireSecs = 1 * 86400;
+        $expireTS = time() + $expireSecs;
+        $cookie = SetCookie::thatExpiresIn('name', 'value', $expireStr);
+        $expected = sprintf('name=value; expires=%s', gmdate(self::$HTTP_DATE_FORMAT, $expireTS));
+        $this->assertEquals($expected, $cookie->toHeaderValue());
+
+        // test negative string expression interval
+        $expireStr = '-1 day';
+        $expireSecs = -1 * 86400;
+        $expireTS = time() + $expireSecs;
+        $cookie = SetCookie::thatExpiresIn('name', 'value', $expireStr);
+        $expected = sprintf('name=value; expires=%s', gmdate(self::$HTTP_DATE_FORMAT, $expireTS));
+        $this->assertEquals($expected, $cookie->toHeaderValue());
+
+        // test 0 string expression interval
+        $expireStr = '0 day';
+        $expireSecs = 0 * 86400;
+        $expireTS = time() + $expireSecs;
+        $cookie = SetCookie::thatExpiresIn('name', 'value', $expireStr);
+        $expected = sprintf('name=value; expires=%s', gmdate(self::$HTTP_DATE_FORMAT, $expireTS));
```

Comment:
These tests can potentially fail? Edge case but still... https://blog.frankdejonge.nl/being-in-control-of-time-in-php/ might be a solution, although that's a bit too much for this library...

Reply:
hi, could you give me an example to see if we can find a non-complicated solution?

Reply:
I've been thinking about this (to solve it in a simple way). I prefer to avoid flickering unit tests. I'm not sure yet what the best solution is. I'll come back to you soon! Thanks again for your work.

Reply:
sure, no problem! :-)
I realized that maybe you were referring to `time()` calls, that can happend in different seconds even if just 1 microsecond apart. In zend-expressive I had to add a double check for min and max values range. The probability of these errors in not 0 and grows bigger with code complexity.

Reply:
just as an example in 1 of my commits:
https://github.com/zendframework/zend-expressive-session-ext/commit/5ebdb3cb1da0db9aed3efcfdf2e572bcafd381be
but here the situation is simpler as i check timestamps, in a cookie-av expire value there is potentially a difference in the resulting day/month/year for just 1 microseconds...and here we have names...not nice values....

Reply:
One possible solution is to add an extractor method in the test cases:
`private function extractExpireTimestamp(string$headerValue): int` and use that value to compare against min/max timestamp values. That method could also contain assertion about existence and expected string format of the expire cookie-av value.

Reply:
...somethig like this:
```php
    private function extractExpireTimestampFrom(string $headerValue): int
    {
        static $expireRegex
            = '/'
            . '(Sun|Mon|Tue||Wed|Thu|Fri|Sat)\,'
            . ' [1-3][0-9]'
            . ' (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)'
            . ' [0-9]{4}'
            . ' [0-2][0-9]\:[0-5][0-9]:[0-5][0-9]'
            . ' [A-Z]{3}'
            . '/';

        $expireString = null;

        $parts = explode(';', $headerValue);
        foreach ($parts as $part) {
            $av = explode('=', $part);
            $avName = strtolower(trim($av[0]));
            if ($avName === 'expires') {
                $expireString = $av[1] ?? null;
                break;
            }
        }

        $this->assertNotNull($expireString);
        $this->assertRegExp($expireRegex, $expireString);

        $expireTimestamp = strtotime($expireString);

        $this->assertNotFalse($expireTimestamp);

        return $expireTimestamp;
    }

```